plugins/processors/execd/execd_test.go

```diff
@@ -0,0 +1,91 @@
+package execd
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestExternalProcessorWorks(t *testing.T) {
+
+	e := New()
+	e.Command = []string{shell(), fileShellScriptPath()}
+	e.RestartDelay = config.Duration(5 * time.Second)
+
+	acc := &testutil.Accumulator{}
+
+	require.NoError(t, e.Start(acc))
+
+	now := time.Now()
+	metrics := []telegraf.Metric{}
+	for i := 0; i < 10; i++ {
+		m, err := metric.New("test",
+			map[string]string{
+				"city": "Toronto",
+			},
+			map[string]interface{}{
+				"population": 6000000,
+				"count":      1,
+			},
+			now)
+		require.NoError(t, err)
+		metrics = append(metrics, m)
+		now = now.Add(1)
+
+		e.Add(m, acc)
+	}
+
+	acc.Wait(1)
+	m := acc.Metrics[0]
```

Comment:
Use GetTelegrafMetrics(), I'm planning to remove the testutil.Metric in my Starlark branch.

I suggest setting up the expected metrics when you create the input metrics, then just calling `testutil.RequireMetricsEqual`